apps/web/pages/settings/my-account/out-of-office/index.tsx

```diff
@@ -86,26 +115,68 @@ const CreateOutOfOfficeEntryModal = ({
     },
   });
 
+  const resetForm = useCallback(() => {
+    reset();
+    setSelectedReason(null);
+    setSelectedMember(null);
+    setProfileRedirect(false);
+  }, [reset]);
+
   const createOutOfOfficeEntry = trpc.viewer.outOfOfficeCreate.useMutation({
     onSuccess: () => {
-      showToast(t("success_entry_created"), "success");
+      showToast(
+        currentlyEditingOutOfOfficeEntry
+          ? t("success_edited_entry_out_of_office")
+          : t("success_entry_created"),
+        "success"
+      );
       utils.viewer.outOfOfficeEntriesList.invalidate();
-      setProfileRedirect(false);
+      resetForm();
       closeModal();
     },
     onError: (error) => {
       showToast(t(error.message), "error");
     },
   });
 
-  const { data: outOfOfficeReasonList } = trpc.viewer.outOfOfficeReasonList.useQuery();
+  useEffect(() => {
+    if (currentlyEditingOutOfOfficeEntry === null) {
+      resetForm();
+      return;
+    }
 
-  const reasonList = [
-    ...(outOfOfficeReasonList || []).map((reason) => ({
-      label: `${reason.emoji} ${reason.userId === null ? t(reason.reason) : reason.reason}`,
-      value: reason.id,
-    })),
-  ];
+    const selectedReason =
+      reasonList.find((reason) => {
+        return reason.value === currentlyEditingOutOfOfficeEntry.selectedReason;
+      }) ?? null;
+    if (selectedReason && selectedReason.value) {
+      setValue("reasonId", selectedReason.value);
```

Comment:
Imo the whole useEffect is unnecessary. Why set default to null only to update it inside the effect. You can use default value as `currentlyEditingOutOfOfficeEntry` instead of updating it inside the effect. In case you are not editing then all values can be null.